includes/abstracts/abstract-wc-product.php

```diff
@@ -1537,16 +1609,22 @@ protected function update_post_meta() {
 			}
 		}
 
-		if ( $this->is_on_sale() ) {
-			update_post_meta( $this->get_id(), '_price', $this->get_sale_price() );
-		} else {
-			update_post_meta( $this->get_id(), '_price', $this->get_regular_price() );
+		if ( in_array( 'regular_price', $updated_props ) || in_array( 'sale_price', $updated_props ) ) {
```

Comment:
Shouldn't we also check for date settings changing here too? sale_price could be previously set, but someone could change and move up the sale date to 'today'.